### DISCUSSION TITLE
Modify LLM prompts

### DISCUSSION THREAD

**Discussion by saurabhsbora** [vanna-ai/vanna#742] [Q&A]
Is it possible to modify the llm prompts to add any custom instructions ?

**Comment by a29874750a** [comment]
Yes, you can go to vanna package and find base.py
the path is vanna/base/base.py
There's a initial_prompt and you can modify it.

**Reply by saurabhsbora** [reply]
yeah did that eventually, but wanted to check if there is any parameter exposed to pass it

**Comment by FatK1234** [comment]
Correct me if I am wrong, but I think we can modify the prompt directly by tweaking the initial_prompt within config? Below is an example.

```
from vanna.ollama import Ollama
from vanna.chromadb import ChromaDB_VectorStore

class MyVanna(ChromaDB_VectorStore, Ollama):
    def __init__(self, config=None):
        ChromaDB_VectorStore.__init__(self, config=config)
        Ollama.__init__(self, config=config)

vn = MyVanna(config={'model': 'gemma3:latest', 'initial_prompt': f"You are a T-SQL / Microsoft SQL Server expert. " + \
            "Generate a SQL query to answer the question."})
```

**Comment by MSchutt** [comment]
With Vanna 2, this works differently.
You can provide a custom System Prompt Class, allowing you to customize every aspect of the prompt. You also have access to the resolved User object, which means you can tailor the prompt for each individual user or user group.
https://vanna.ai/docs/placeholder/system-prompts
